scarb/src/compiler/compilation_unit.rs

```diff
@@ -92,6 +96,20 @@ pub struct CompilationUnitCairoPlugin {
     pub builtin: bool,
 }
 
+/// Unique identifier of the compilation unit component.
+/// Currently, a compilation unit can be uniquely identified by [`PackageId`] only.
+/// It may be not sufficient in the future depending on changes to the compilation model.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct CompilationUnitComponentId {
+    pub package_id: PackageId,
+}
+
+impl CompilationUnitComponentId {
+    pub fn to_discriminator(&self) -> String {
+        self.package_id.to_serialized_string()
+    }
```

Comment:
Changed discriminator to be serialized `PackageId` instead of `Version` to be aligned with the comment to `CompilationUnitComponentDependencyMetadata::id`

Reply:
This requires changes to Cairo to work ^

Reply:
I mean `ProjectConfig` is a terrible structure that sometimes has duplicated information, but sometimes lacks it (like in this case), so rn in `RootDatabaseBuilder` we set the discriminator to always be equal to `CrateSetting.version`

https://github.com/starkware-libs/cairo/blob/932767340c4b8a762140bf9eba305f437587ac1b/crates/cairo-lang-compiler/src/project.rs#L148-L152